spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -220,8 +220,20 @@ object Engine {
   // Thread pool for layer use
   @volatile private var _model: ThreadPool = new ThreadPool(1)
 
-  // Thread pool for read data
-  @volatile private var _io: ThreadPool = null
+  // This thread is mainly for mkldnn library.
+  // Because if we use the parent thread directly, there will be two bugs,
+  //   1. The child threads forked from parent thread will be bound to core 0
+  //      because of the affinity settings.
+  //   2. The native thread has some unknown thread local variables. So if
+  //      the parent thread exits and is recreated, such as the thread from
+  //      Executors.newFixedThreadPool. The whole app will be segment fault.
+  // The parent thread means the main thread (Local Mode) or worker thread of
+  // `mapPartition` (Distributed Mode).
+  // --------------------------------------------------------------------------
+  // We will only use the `threadPool` in ThreadPool, which is a ExecutorService.
+  // For `context` in ThreadPool, it is the called thread when poolSize is 1.
+  // So many usages of that thread, we will not change it for now.
+  val computing: ThreadPool = new ThreadPool(1)
```

Comment:
maybe give more clear name to illustrate this thread is just for mkldnn model.

Reply:
agreed, like `dnnComputing` or just `dnn`

Reply:
OK.